### PR TITLE
add initial playwright test

### PR DESCRIPTION
**Pull request by jamesgorrie**
# What's changed
- adds an initial playwright e2e test

[🍏 Here's a successful run](https://github.com/climatepolicyradar/navigator-frontend/actions/runs/11957984101/job/33336425619?pr=288)

[🔴 Here's broken run](https://github.com/climatepolicyradar/navigator-frontend/actions/runs/11938826616/job/33277946517)

[📜 Here's some nice docs on how to debug the broken run](https://playwright.dev/docs/ci-intro#html-report)

There is a few areas where this significantly differs from [the current implementation of e2e tests](https://github.com/climatepolicyradar/navigator-frontend/tree/main/e2e). Here's the thinking around it

- **The e2e test are now part of the `src` app.**
  - Yup. This is partially because of co-location reasons. We try to keep the tests as close to the implementation as possible to make them less of an after thought.
  - It's easier to just start the app as they're co-located
  - This does make it less modular though, but haven't seen an opportunity where composition would be useful, so this might be a premature abstraction, but could be missing things
- **Playwright?**
  - after talking to @annaCPR and @PatFawbertMills there did not seem to be any reason we went with Cypress bar that it's there. I have done a few migrations recently from Cypress => Playwright, so I am biased, but those were driven by cypress being a little more flaky, and playwright's ecosystem being far more rich due to the engineering effort going into it (thanks M$)
- **A new actions?** This is to keep this PR clear in it's inception, with the desire to bring it into the [ci/cd pipeline](https://github.com/climatepolicyradar/navigator-frontend/blob/main/.github/workflows/ci-cd.yml). The current e2e tests are also located in the `theme-factory` action, which is coupled to this repo, so we're making that coupling more explicit by keeping the code that tests this code here

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

### CODE REVIEW

File: tests/search.spec.ts
Comment (jamesgorrie) [inline tests/search.spec.ts:1]: this is based on [a manual test run we have used previously 🔒](https://www.notion.so/climatepolicyradar/Manual-test-plan-69c302285d284598b7571d97134b7b45)

**Comment by jamesgorrie** [comment]
Here's a successful run.

File: .env.example [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
-ADOBE_API_KEY=1234567890
+ADOBE_API_KEY=dca9187b65294374a6367824df902fdf
 
 REDIRECT_FILE="test.json"
 
-HOSTNAME=https://localhost:3000
+HOSTNAME=http://localhost:3000
 
-NEXT_PUBLIC_APP_TOKEN=some_token
+NEXT_PUBLIC_APP_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhbGxvd2VkX2NvcnBvcmFfaWRzIjpbIkNDTFcuY29ycHVzLmkwMDAwMDAwMS5uMDAwMCIsIkNQUi5jb3JwdXMuaTAwMDAwMDAxLm4wMDAwIiwiVU5GQ0NDLmNvcnB1cy5pMDAwMDAwMDEubjAwMDAiXSwiZXhwIjoyMDQyMTEzMzY5LCJpYXQiOjE3MjY1NzY5NjksImlzcyI6IkNsaW1hdGUgUG9saWN5IFJhZGFyIiwic3ViIjoiQ1BSIiwiYXVkIjoiaHR0cHM6Ly9hcHAuZGV2LmNsaW1hdGVwb2xpY3lyYWRhci5vcmcvIn0.mJ2qLJmMyPLGt0rM_tTXhlVv1glxooxmQV0bWrvPwKU
```
Comment (jamesgorrie) [inline .env.example:15]: [These are already exposed here](https://github.com/climatepolicyradar/navigator-frontend/blob/main/Makefile#L3-L8).

File: .env.example [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
```
Comment (annaCPR) [inline .env.example:15]: Are we ok committing staging secrets to version control?

File: tests/search.spec.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { test, expect } from "@playwright/test";
+
+test("search", async ({ page }) => {
+  /** Reject the consent banner */
+  await page.goto("/");
+  await page.getByRole("button", { name: "Reject" }).click();
```
Comment (annaCPR) [inline tests/search.spec.ts:6]: Can we add an assertion that the heading of the cookie banner is visible? Just to make it clearer as to what we're rejecting.

File: .env.example (hunk 2 above)
Comment (jamesgorrie) [reply .env.example:15]: Not 100% sure, [but these are exposed here](https://github.com/climatepolicyradar/navigator-frontend/pull/288#discussion_r1852520087).

File: src/components/search/SearchResultList.tsx [hunk 4]
```diff
@@ -58,13 +58,14 @@ const SearchResultList = ({ category, families, activeFamilyIndex, onClick }: TP
     return <div className="h-96 mt-4 md:mt-0">Your search returned no results.</div>;
   }
   return (
-    <>
+    <div>
+      <h2 className="sr-only">Search results</h2>
```
[the remark this answers is not in the document]
Comment (jamesgorrie) [reply src/components/search/SearchResultList.tsx:62]: Mostly, but also accessibility. [Headings are used by screen-readers to navigate the page structure](https://www.a11yproject.com/posts/how-to-accessible-heading-structure/) - this seems like a natural place a person might want to skip to.

This is often a nice byproduct of these tests - in this case, surfacing that we don't sign post what the proceeding piece of content is defined as. We could probably make better use of sections etc.

I thought defining the [accessibility tree](https://developer.mozilla.org/en-US/docs/Glossary/Accessibility_tree) was outside the scope of this PR - but if we think it important to address here, we can.

File: .env.example (hunk 1 above)
Comment (annaCPR) [reply .env.example:15]: Ok, didn't know that. Just wanted to make sure we're all aware and agreed on this.

File: .env.example (hunk 2 above)
Comment (annaCPR) [reply .env.example:15]: Ok, didn't know that. Just wanted to make sure we're all aware and agreed on this.

File: src/components/search/SearchResultList.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (annaCPR) [reply src/components/search/SearchResultList.tsx:62]: 👍  I'm definitely pro using more semantic HTML and making our apps more accessible - there's definitely room for improvement there. Might be worth discussing the topic with the wider group.

**Pull request merged**